app/packages/core/src/components/Filters/categoricalFilter/Wrapper.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { MutableRefObject } from "react";
+import {
+  RecoilState,
+  useRecoilState,
+  useRecoilValue,
+  useSetRecoilState,
+} from "recoil";
+
+import * as fos from "@fiftyone/state";
+
+import FilterOption from "./filterOption/FilterOption";
+import Checkbox from "../../Common/Checkbox";
+import { Button } from "../../utils";
+import { CHECKBOX_LIMIT, nullSort } from "../utils";
+import { isKeypointLabel, V } from "./CategoricalFilter";
+
+interface WrapperProps {
+  results: [V["value"], number][];
+  selectedValuesAtom: RecoilState<V["value"][]>;
+  excludeAtom: RecoilState<boolean>;
+  isMatchingAtom: RecoilState<boolean>;
+  onlyMatchAtom: RecoilState<boolean>;
+  color: string;
+  totalCount: number;
+  modal: boolean;
+  path: string;
+  selectedCounts: MutableRefObject<Map<V["value"], number>>;
+}
+
+const Wrapper = ({
+  color,
+  results,
+  totalCount,
+  selectedValuesAtom,
+  excludeAtom,
+  isMatchingAtom,
+  onlyMatchAtom,
+  modal,
+  path,
+  selectedCounts,
+}: WrapperProps) => {
+  const name = path.split(".").slice(-1)[0];
+  const schema = useRecoilValue(fos.field(path));
+  const [selected, setSelected] = useRecoilState(selectedValuesAtom);
+  const selectedSet = new Set(selected);
+  const setExcluded = excludeAtom ? useSetRecoilState(excludeAtom) : null;
+  const setOnlyMatch = onlyMatchAtom ? useSetRecoilState(onlyMatchAtom) : null;
+  const setIsMatching = isMatchingAtom
+    ? useSetRecoilState(isMatchingAtom)
+    : null;
+  const sorting = useRecoilValue(fos.sortFilterResults(modal));
+  const counts = Object.fromEntries(results);
+  let allValues: V[] = selected.map<V>((value) => ({
+    value,
+    count: counts[String(value)] ?? 0,
+  }));
+  const skeleton = useRecoilValue(isKeypointLabel(path));
+  const neverShowExpansion = schema?.ftype.includes("ObjectIdField");
+
+  if ((results.length <= CHECKBOX_LIMIT && !neverShowExpansion) || skeleton) {
+    allValues = [
+      ...allValues,
+      ...results
+        .filter(([v]) => !selectedSet.has(v))
+        .map(([value, count]) => ({ value, count })),
+    ];
+  }
+
+  if (totalCount === 0) {
+    return (
```

Comment:
This is not allowed. It will change the number hooks rendered.  The simplest fix will be to put after all hooks, i.e. `use*` calls though it is not the most performant.

<img width="1470" alt="Screen Shot 2023-01-13 at 9 06 58 AM" src="https://user-images.githubusercontent.com/19821840/212352382-7eb39baa-b24b-415f-b2de-8f8fe7c01486.png">